ibllib/io/video.py

```diff
@@ -50,6 +53,16 @@ def get_video_frames_preload(video_path, frame_numbers, mask=Ellipsis, as_list=F
     cap = VideoStreamer(video_path).cap if is_url else cv2.VideoCapture(str(video_path))
     assert cap.isOpened(), 'Failed to open video'
 
+    frame_count = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+    frame_numbers = frame_numbers if frame_numbers is not None else range(frame_count)
+    # Check that frame numbers don't exceed video size, else remove the frames that do
+    if max(frame_numbers) >= frame_count:
+        print(f'Frame numbers exceed frame count, removing frames above {frame_count}')
```

Comment:
What's the reason you don't raise a ValueError?

Reply:
The program can still run, it is more of  a warning rather than error, so I handled it similarly to the one in line 84 of this function. 
I implemented this so that I can loop over a set number of frames (e.g. 100 at a time) and when there are not enough frames left at the end of the video, it will just return the frames that exist, rather than returning a bunch of zero arrays

Reply:
I suppose it's fine, but it means the user has to size checks outside the function to avoid dimension mismatches errors further down the line.

Reply:
Yeah true. For computing the motion energy it was a bit annoying if it returned frames that didn't exist, because it meant I didn't have a way to know when the reading of the actual video was finished. The alternative would be to access the metadata of the video to find out how many frames there are before calling this function. But somehow it didn't seem super elegant to access the video twice. I am fine to go either way.

Reply:
The initial idea was to load the video timestamps (camera.times) first and use that to determine which and how many frames to load.

Reply:
Ok then, will do that